www/js/lib/zimfile.js

```diff
@@ -186,16 +186,30 @@ define(['xzdec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'], function(xz, util,
         {
             var clusterOffset = readInt(clusterOffsets, 0, 8);
             var nextCluster = readInt(clusterOffsets, 8, 8);
+            // This method of calculating cluster size is not safe: see https://github.com/openzim/libzim/issues/84#issuecomment-612962250
+            // var thisClusterLength = nextCluster - clusterOffset - 1;
```

Comment:
This is a note to a dev (mysefl) to remind me  that we should not use the apparently simple method of feeding a whole compressed cluster to the decompressor. There are some reasons why it is not safe to calculate cluster length by substracting the next pointer from the previous pointer. In any case, streaming means we can abandon a decompression if the blob we want is near the beginning of the cluster. The only advantage in decompressing a full cluster (which is what libzim does) would be if we had a cluster cache like the one written by peter-x. Then there might be performance gains if one cluster contains several blobs that we want belonging to an article. We would only decompress the cluster once, instead of starting again from the beginning for each blob. Swings and roundabouts...